frontend/src/utils/filterOrders.ts

```diff
@@ -0,0 +1,77 @@
+import Order from '../models/Order.model';
+
+interface BaseFilter {
+  currency: number;
+  type: number;
+}
+
+interface AmountFilter {
+  amount: string;
+  minAmount: string;
+  maxAmount: string;
+  threshold: number;
+}
+
+interface FilterOrders {
+  order: Order;
+  baseFilter: BaseFilter;
+  amountFilter?: AmountFilter | null;
+  paymentMethods?: string[];
+}
+
+const filterByPayment = function (order: Order, paymentMethods: string[]) {
+  if (paymentMethods.length === 0) {
+    return true;
+  } else {
+    let result = false;
+    paymentMethods.forEach((method) => {
+      result = result || order.payment_method.includes(method.name);
+    });
+    return result;
+  }
+};
+
+const filterByAmount = function (order: Order, filter: AmountFilter) {
+  if (filter.amount != '') {
+    if (order.amount === '' || order.amount === null) {
+      return (
+        order.max_amount < filter.amount * (1 + filter.threshold) &&
+        order.min_amount > filter.amount * (1 - filter.threshold)
+      );
+    } else {
+      return (
+        order.amount < filter.amount * (1 + filter.threshold) &&
+        order.amount > filter.amount * (1 - filter.threshold)
+      );
+    }
+  } else {
+    if (order.amount === '' || order.amount === null) {
+      return (
+        order.max_amount < filter.maxAmount * (1 + filter.threshold) &&
+        order.min_amount > filter.minAmount * (1 - filter.threshold)
+      );
+    } else {
+      return (
+        order.amount < filter.maxAmount * (1 + filter.threshold) &&
+        order.amount > filter.minAmount * (1 - filter.threshold)
+      );
+    }
+  }
+};
```

Comment:
I think that the final logic should be the same in all scenarios right? the only change is the source, not sure if I did well but maybe something like this might help for future maintenance

````
const filterByAmount = function (order: Order, filter: AmountFilter) {
    const filterMaxAmount = filter.amount != '' ? filter.amount : filter.maxAmount
    const filterMinAmount = filter.amount != '' ? filter.amount : filter.minAmount
    const orderMinAmount = order.amount === '' || order.amount === null ? order.min_amount : order.amount
    const orderMaxAmount = order.amount === '' || order.amount === null ? order.max_amount : order.amount

    return (
        orderMaxAmount < filterMaxAmount * (1 + filter.threshold) &&
        orderMinAmount > filterMinAmount * (1 - filter.threshold)
    );
};
````

Reply:
This is certainly much better. My thinking was not clear and the code reflected it :sweat_smile: . Thanks a lot!